### PR TITLE
Adds shutdown server admin verb.

### PR DESCRIPTION
**Pull request by MrStonedOne**
Requires the server be at pregame or endround.

Requires the start/end be delayed

Requires two confirmations from the same admin, 30 seconds apart.

During the first 30 second delay, all admins are notified and alerted and those with +SERVER can cancel the process within those 30 seconds

After that final confirmation, it announces the shutdown to players, and waits another 30 seconds where admins can cancel the shutdown still.

Requires the `ALLOW_SHUTDOWN` config. This config is protected from runtime editing via VV.

image

### CODE REVIEW

File: code/modules/admin/server_verbs.dm [hunk 1]
```diff
@@ -17,6 +17,87 @@
...
+		return
+
+	to_chat(world, "<span class='danger'>Server shutting down.</span> <span class='notice'>Initiated by: [shuttingdown]</span>")
+
+	if (GLOB.tgs)
+		var/datum/tgs_api/TA = GLOB.tgs
+		TA.EndProcess()
+
+	sleep(world.tick_lag)
+	qdel(world) //there are a few ways to shutdown the server, but this is by far my favorite
```
Comment (MrStonedOne) [inline code/modules/admin/server_verbs.dm]: ^^^

File: code/modules/admin/server_verbs.dm [hunk 2]
```diff
@@ -17,6 +17,87 @@
...
+		return
+
+	if (shuttingdown != usr.ckey) //somebody cancelled but then somebody started again.
+		return
+
+	to_chat(world, "<span class='danger'>Server shutting down.</span> <span class='notice'>Initiated by: [shuttingdown]</span>")
+
+	if (GLOB.tgs)
+		var/datum/tgs_api/TA = GLOB.tgs
+		TA.EndProcess()
```
Comment (MrStonedOne) [inline code/modules/admin/server_verbs.dm]: @Cyberboss this restarts the server, it doesn't shut it down like the name implies.

What should I put here if I want to shut the server down?

File: config/config.txt [hunk 3]
```diff
@@ -21,6 +21,17 @@ HUBPASSWORD kMZy3U5jJHSiBQjr
 ## Add a # infront of this if you want to use the SQL based admin system, the legacy system uses admins.txt. You need to set up your database to use the SQL based system.
 ADMIN_LEGACY_SYSTEM
 
+## Add a # infront of this to disable the shutdown server verb.
+ALLOW_SHUTDOWN
```
Comment (Rohesie) [inline config/config.txt]:
```suggestion
##ALLOW_SHUTDOWN
```
This should be disabled by default. It's more likely for downstreams to not be using TGS3.

Comment (MrStonedOne) [reply config/config.txt]: 1 config file defaults are aimed at development defaults, not production defaults.

2 this does not need tgs to work, it just needs snowflake if you have tgs.

Comment (Rohesie) [reply config/config.txt]: Fair

**Pull request merged**